File: cpp/TwoSum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        vector<int> res;
        map<int, int> map_nums;
        for(int i=0;i<nums.size();++i){
            map_nums[nums[i]]=i;
        }
        for(int i=0;i<nums.size();++i){
            const int gap = target - nums[i];
            if (map_nums.find(gap)!= map_nums.end() && i!=map_nums[gap]) {
                  res.push_back(i);
                  res.push_back(map_nums[gap]);
                  break;
            }
        }
        return res;
    }
};

File: cpp/Two_Sum.java
public class Solution {
    public int[] twoSum(int[] numbers, int target) {
      int[] res=new int[2];
      HashMap<Integer,Integer> nums=new HashMap<Integer,Integer>();
      for(int i=0;i<numbers.length;i++){
          nums.put(numbers[i],i);
      }
      for(int i=0;i<numbers.length;i++){
          if(nums.containsKey(target - numbers[i]) && i != nums.get(target - numbers[i])){
              res[0]=i;
              res[1]=nums.get(target - numbers[i]);
          }
      }
      return res;
    }
}
